$OpenBSD: patch-mk_bootstrap_mk,v 1.5 2007/07/21 17:14:58 kili Exp $

Don't use -fno-unit-at-a-time on amd64 when bootstrapping from HC files.
Pull in our system libgmp.
libreadline needs libncurses.

--- mk/bootstrap.mk.orig	Wed Apr 25 19:10:40 2007
+++ mk/bootstrap.mk	Sat Jun  2 14:01:38 2007
@@ -22,7 +22,7 @@ PLATFORM_HC_BOOT_CC_OPTS += -fno-defer-pop -fomit-fram
 endif
 
 ifeq "$(x86_64_TARGET_ARCH)" "1"
-PLATFORM_HC_BOOT_CC_OPTS += -fomit-frame-pointer -fno-asynchronous-unwind-tables -fno-unit-at-a-time -fno-builtin
+PLATFORM_HC_BOOT_CC_OPTS += -fomit-frame-pointer -fno-asynchronous-unwind-tables -fno-builtin
 endif
 
 ifeq "$(hppa_TARGET_ARCH)" "1"
@@ -83,9 +83,11 @@ endif
 
 ifeq "$(HaveLibGmp)" "NO"
 DASH_L_GHC_RTS_GMP_DIR=-L$(FPTOOLS_TOP_ABS)/$(GHC_RTS_DIR_REL)/gmp
+else
+DASH_L_GHC_RTS_GMP_DIR=-L$(GMP_LIB_DIRS)
 endif
 
-HC_BOOT_LD_OPTS =				\
+HC_BOOT_LD_OPTS +=				\
    -L$(FPTOOLS_TOP_ABS)/$(GHC_RTS_DIR_REL)	\
    $(DASH_L_GHC_RTS_GMP_DIR)                    \
    -L$(FPTOOLS_TOP_ABS)/libraries/base		\
@@ -142,7 +144,7 @@ HC_BOOT_LD_OPTS += \
 HC_BOOT_LIBS =
 
 ifeq "$(GhcWithInterpreter)" "YES"
-HC_BOOT_LIBS += -lHSreadline -lreadline -lHStemplate-haskell -lHSunix -lHSunix_cbits
+HC_BOOT_LIBS += -lHSreadline -lreadline -lcurses -lHStemplate-haskell -lHSunix -lHSunix_cbits
 endif
 
 HC_BOOT_LIBS +=  -lHSCabal -lHShaskell98 -lHSregex-compat -lHSregex-posix -lHSregex-base -lHSbase -lHSbase_cbits -lHSparsec -lHSrts -lgmp -lm $(EXTRA_HC_BOOT_LIBS)
